web-client/src/ducks/auth/reducer.ts

```diff
@@ -20,11 +28,93 @@ export default createReducer<AuthState>(
     ) => {
       state.token = payload.accessToken;
     },
-    [FIREBASE_FACEBOOK_LOGIN.COMPLETED]: (
+    [FIREBASE_FACEBOOK_LOGIN_POPUP.PENDING]: (state: AuthState) => ({
+      ...state,
+      actionInProgress: true,
+    }),
+    [FIREBASE_FACEBOOK_LOGIN_POPUP.COMPLETED]: (
       state: AuthState,
-      { payload }: { payload: firebase.auth.UserCredential },
+      { payload }: { payload: string },
     ) => {
-      state.token = (payload.credential as firebase.auth.OAuthCredential).accessToken;
+      const token = payload;
+      const actionInProgress = false;
+      return {
+        ...state,
+        token,
+        actionInProgress,
+      };
+    },
+    [FIREBASE_FACEBOOK_LOGIN_POPUP.REJECTED]: (
+      state: AuthState,
+      { payload }: { payload: Error },
+    ) => {
+      const error = payload;
+      const actionInProgress = false;
+      return {
+        ...state,
+        error,
+        actionInProgress,
+      };
+    },
+    [FIREBASE_FACEBOOK_LOGIN_REDIRECT_START.PENDING]: (state: AuthState) => {
+      window.localStorage.setItem('redirect_started', new Date().toISOString());
+      const actionInProgress = true;
+      return {
+        ...state,
+        actionInProgress,
+      };
+    },
+    [FIREBASE_FACEBOOK_LOGIN_REDIRECT_START.REJECTED]: (
+      state: AuthState,
+      { payload }: { payload: Error },
+    ) => {
+      window.localStorage.removeItem('redirect_started');
+      const error = payload;
+      const actionInProgress = false;
+      return {
+        ...state,
+        error,
+        actionInProgress,
+      };
+    },
+    [FIREBASE_FACEBOOK_LOGIN_REDIRECT_START.COMPLETED]: (state: AuthState) => {
+      const actionInProgress = false;
+      return {
+        ...state,
+        actionInProgress,
+      };
+    },
+    [FIREBASE_FACEBOOK_LOGIN_REDIRECT_COMPLETE.PENDING]: (state: AuthState) => {
+      window.localStorage.removeItem('redirect_started');
```

Comment:
Here we could just save `redirect_started` in reducer state perhaps?

Reply:
my intention for redirect_started as a parameter in "localStorage" was for it to persist across browser sessions.
We can use this in the manner you are suggesting only if we are using redux-persist, which in my opinion shouldn't be used for the purpose of solving this particular issue only. and should be for improving user experience offline.
However as I have mentioned in the previous comment, this my opinion.
I am open to your point of view and am willing to add redux-persist but that would be done as a separate PR